internal/pkg/cli/pipeline_list.go

```diff
@@ -75,60 +107,157 @@ func (o *listPipelineOpts) Ask() error {
 		}
 		o.appName = app
 	}
+
 	return nil
 }
 
 // Execute writes the pipelines.
 func (o *listPipelineOpts) Execute() error {
-	var out string
-	pipelines, err := o.pipelineLister.ListDeployedPipelines(o.appName)
+	ctx, cancel := context.WithTimeout(context.Background(), pipelineListTimeout)
+	defer cancel()
+
+	switch {
+	case o.shouldShowLocalPipelines && o.shouldOutputJSON:
+		return o.jsonOutputLocal(ctx)
+	case o.shouldShowLocalPipelines:
+		return o.humanOutputLocal()
+	case o.shouldOutputJSON:
+		return o.jsonOutputDeployed(ctx)
+	}
+
+	return o.humanOutputDeployed()
```

Comment:
How about this
```suggestion
         if o.shouldOutputJSON {
                  ctx, cancel := context.WithTimeout(context.Background(), pipelineListTimeout)
                  defer cancel()
                  if o.shouldShowLocalPipelines {
                           return o.jsonOutputLocal(ctx)
                  }
                  return o.jsonOutputDeployed(ctx)
         }
         if o.shouldShowLocalPipelines {
                  return o.humanOutputLocal()
         }
         return o.humanOutputDeployed()
```

Reply:
I'm don't think there's much overhead in creating a context (plus, ideally, it'd be nice if `humanOutputDeployed()` used one to call `ListDeployedPipelines()`), so I personally would rather have less nesting and keep it as is, but if you think the other option is better to read I'm fine switching.